### PR TITLE
Key Gear: Auto-add current device

### PR DESCRIPTION
**Pull request by hedvigoscar**
This PR is not mergeable until the backend can support it properly.

### CODE REVIEW

**Comment by hedvigoscar** [comment]
Note to reviewer: the mutation `createKeyGearItem` is idempotent on `physicalReferenceHash`.

**Comment by hedvigoscar** [comment]
This PR is now mergeable.

File: app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    @SuppressLint("HardwareIds")
+    fun getDeviceFingerprint(): String {
+        val messageDigest = MessageDigest.getInstance("SHA-256")
+        messageDigest.reset()
+
+        val androidId = Settings.Secure.getString(context.contentResolver, Settings.Secure.ANDROID_ID)
+        messageDigest.update(androidId.toByteArray())
+
+        val result = messageDigest.digest()
+        return BigInteger(1, result).toString(16).padStart(32, '0')
```
Comment (FredrikAreschoug) [inline app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt]: why do we do this?

File: app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+        windowManager.defaultDisplay.getRealMetrics(displayMetrics)
+
+        val widthInches = displayMetrics.widthPixels / displayMetrics.xdpi
+        val heightInches = displayMetrics.heightPixels / displayMetrics.xdpi
+
+        val diagonalInches = sqrt(widthInches.pow(2) + heightInches.pow(2))
+
+        // This is unfortunately the only trick that is available
+        if (diagonalInches >= 7.0f) {
```
Comment (FredrikAreschoug) [inline app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt]: :(

Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt]: I know 🙁

File: app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt (hunk 1 above)
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt]: This is the best way I found to convert it from a byte array into a hex string 🙂

File: app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import java.security.MessageDigest
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+class DeviceInformationService(
+    private val context: Context
+) {
+    @SuppressLint("HardwareIds")
+    fun getDeviceFingerprint(): String {
+        val messageDigest = MessageDigest.getInstance("SHA-256")
```
[the remark this answers is not in the document]
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt]: I checked and was unable to find it 🙁

**Pull request merged**